spec/server/update_model_spec.rb

```diff
@@ -0,0 +1,87 @@
+describe UpdateModelController do
+  include Rack::Test::Methods
+
+  def app
+    OUTER_APP
+  end
+
+  let(:proper_name) { 'name' }
+  let(:improper_name) { 'namex' }
+
+  it "rejects requests from non-superusers" do
+    auth_header(:editor)
+
+    json_post(:update_model, {
+      project_name: "labors",
+      actions: [
+        action_name: "update_attribute",
+        model_name: "monster",
+        attribute_name: "name",
+        description: "The monster's name",
+      ]
+    })
+
+    expect(last_response.status).to eq(403)
+  end
+
+  it "updates attribute options" do
+    auth_header(:superuser)
+    json_post(:update_model, {
+      project_name: "labors",
+      actions: [{
+        action_name: "update_attribute",
+        model_name: "monster",
+        attribute_name: proper_name,
+        description: "The monster's name",
+        display_name: "NAME"
+      }]
+    })
+
+    expect(last_response.status).to eq(200)
+    expect(Labors::Monster.attributes[:name].description).to eq("The monster's name")
+    expect(Labors::Monster.attributes[:name].display_name).to eq("NAME")
+
+    response_json = JSON.parse(last_response.body)
+    attribute_json = response_json["models"]["monster"]["template"]["attributes"]["name"]
+    expect(attribute_json["desc"]).to eq("The monster's name")
+    expect(attribute_json["display_name"]).to eq("NAME")
+  end
+
+  it "does not update attribute options with invalid attribute name" do
+    auth_header(:superuser)
+    json_post(:update_model, {
+      project_name: "labors",
+      actions: [{
+        action_name: "update_attribute",
+        model_name: "monster",
+        attribute_name: improper_name,
+        description: "The monster's name",
+        display_name: "NAME"
+      }]
+    })
+
+    response_json = JSON.parse(last_response.body)
+    expect(last_response.status).to eq(422)
+    expect(response_json['errors'][0]['message']).to eq('Attribute does not exist')
+  end
+
+
+  it "does not update attribute options with the incorrect data type" do
+    auth_header(:superuser)
+    json_post(:update_model, {
+      project_name: "labors",
+      actions: [{
+        action_name: "update_attribute",
+        model_name: "monster",
+        attribute_name: proper_name,
+        description: 123,
+        display_name: "NAME",
+        hidden: 'something'
+      }]
+    })
+
+    response_json = JSON.parse(last_response.body)
+    expect(last_response.status).to eq(422)
+    expect(response_json['errors'][0]['message']).to eq("Update attribute failed")
```

Comment:
A clearer message here might help the user correct the request.

Reply:
@graft, we capture the error message that comes back from Sequel on the insert failure: 
```
{"message"=>"Update attribute failed",
 "source"=>{"attribute_name"=>"name", "model_name"=>"monster"},
 "reason"=>
  "PG::InvalidTextRepresentation: ERROR:  invalid input syntax for type boolean: \"something\"\nLINE 1: ....024405-0600', '2020-06-11 09:04:33.024406-0600', 'something...\n                                                             ^\n"}
```

I'll update the spec to reflect that.

Reply:
Argh i wrote a whole thing about this earlier but it didn't post? Anywho see above.

Reply:
About rescuing all errors and the necessity of differentiating between a 422 & 500? If so, yes: https://github.com/mountetna/magma/pull/168#discussion_r438913270